security/fma-generic-contracts.md

```diff
@@ -0,0 +1,158 @@
+
+- **Mitigations:**
+  Storage layouts are validated to only extend and not rewrite the slots of the previous version.
+
+- **Detection:**
+  If this error were missed, it could be detected by looking at the storage values in the proxy or recognizing an inconsistency in variables stored in overwritten slots.
+  Alternatively, we could detect it from monitoring or user reports that something seems off/incorrect.
+
+- **Recovery Path(s):**
+  Deploy new implementation with a corrected storage layout and write correct values to storage.
+  Requires an upgrade.
+
+
+#### Proxy Initialized to Wrong Values
+
+- **Description:**
+    The proxy is initialized with the wrong values.
+
+- **Risk Assessment:**
+  High severity since arbitrary values can be incorrectly used.
+  Low likelihood since proxy initialization has a small surface area and is well verified/tested.
+
+- **Mitigations:**
+  Validate initialized values through extensive unit testing.
+  - See: Instances of `test_initialize_succeeds()` such as the one [here](https://github.com/ethereum-optimism/optimism/blob/e6ef3a900c42c8722e72c2e2314027f85d12ced5/packages/contracts-bedrock/test/L1/L1CrossDomainMessenger.t.sol#L37-L44).
+  - See: Deploy script checks incorporated in [ChainAssertions](https://github.com/ethereum-optimism/optimism/commit/e6ef3a900c42c8722e72c2e2314027f85d12ced5#diff-0f78978618d5f98971fee611fc5476e58643902051d969e0616d5d91a05cff9c) when `_isProxy` = `true`.
+  Simulations prior to performing the upgrade, verifying the initialized values.
+
+- **Detection:**
+  A simulation can show an unintended initialized proxy value.
+  Monitoring or user reports.
+
+- **Recovery Path(s):**
+  Repair the upgrade path (if needed).
+  Upgrade the smart contract.
+
+
+#### Can’t Upgrade Implementation
+
+- **Description:**
+  Contracts are unable to be upgraded.
+
+- **Risk Assessment:**
+  High severity since it effectively bricks an arbitrary contract in the system.
+  Low likelihood since contract upgrades are verified and follow industry-wide, standardized patterns.
+
+- **Mitigations:**
+  Validate through extensive unit testing.
+  Perform simulations that demonstrate upgradability.
+
+- **Detection:**
+  Simulate a contract upgrade to check for failure.
+
+- **Recovery Path(s):**
+  Upgrade contracts and/or proxies.
+  Update `op-chain-ops` if needed.
+
+
+#### Able to Reinitialize Implementation
+
+- **Description:**
+  A third party is able to reinitialize the contract to arbitrary values.
+
+- **Risk Assessment:**
+  High severity since an arbitrary actor can perform an action with arbitrary values.
+  Low likelihood since authentication is a critical, tested and validated property of the system.
+
+- **Mitigations:**
+  Validate through extensive unit testing.
+  - See: [Initializable.t.sol](https://github.com/ethereum-optimism/optimism/blob/e6ef3a900c42c8722e72c2e2314027f85d12ced5/packages/contracts-bedrock/test/vendor/Initializable.t.sol).
```

Comment:
There is also `InitializerOZv5.t.sol`, since this is the generic FMA we should mention both

Reply:
I've dug through ctb, but don't see any reference of `InitializerOZv5.t.sol`. Can we add this in a follow-on pr once this lands?

Reply:
Yep, just make sure to track somewhere so we don't forget! 

It's here: https://github.com/ethereum-optimism/optimism/blob/4b1c12ad4312fe6fe28d8297a26cccf053356704/packages/contracts-bedrock/test/vendor/InitializableOZv5.t.sol

Reply:
I've added it, thank you for linking!